src/libstd/sys_common/backtrace.rs

```diff
@@ -39,26 +37,74 @@ pub fn print(w: &mut dyn Write, format: PrintFormat) -> io::Result<()> {
     }
 }
 
-fn _print(w: &mut dyn Write, format: PrintFormat) -> io::Result<()> {
-    writeln!(w, "stack backtrace:")?;
+fn _print(w: &mut dyn Write, format: PrintFmt) -> io::Result<()> {
+    struct DisplayBacktrace {
+        format: PrintFmt,
+    }
+    impl fmt::Display for DisplayBacktrace {
+        fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
+            _print_fmt(fmt, self.format)
+        }
+    }
+    write!(w, "{}", DisplayBacktrace { format })
+}
 
-    let mut printer = Printer::new(format, w);
+fn _print_fmt(fmt: &mut fmt::Formatter<'_>, print_fmt: PrintFmt) -> fmt::Result {
+    let mut print_path = move |fmt: &mut fmt::Formatter<'_>, bows: BytesOrWideString<'_>| {
+        output_filename(fmt, bows, print_fmt)
+    };
+    let mut bt_fmt = BacktraceFmt::new(fmt, print_fmt, &mut print_path);
+    bt_fmt.add_context()?;
+    let mut skipped = false;
     unsafe {
+        let mut idx = 0;
+        let mut res = Ok(());
         backtrace::trace_unsynchronized(|frame| {
+            if idx > MAX_NB_FRAMES {
+                skipped = true;
+                return false;
+            }
+
+            // Remove the `17: 0x0 - <unknown>` line.
+            if print_fmt == PrintFmt::Short && frame.ip() == ptr::null_mut() {
+                skipped = true;
+                return true;
+            }
+
             let mut hit = false;
-            backtrace::resolve_frame_unsynchronized(frame, |symbol| {
-                hit = true;
-                printer.output(frame, Some(symbol));
-            });
+            if cfg!(not(target_os = "fuchsia")) { // Fuchsia can't symbolize in-proc
```

Comment:
Is this clause necessary? While technically doing work that won't do much I think thisll work on fuchsia without the clause?

Reply:
nope, not necessary, just saves some unnecessary work. Happy to take it out if you'd prefer.

Reply:
Yeah if possible I think it's best to keep platform specific config away from here

Reply:
Done.